src/dmd/mars.d

```diff
@@ -643,6 +643,22 @@ private int tryMain(size_t argc, const(char)** argv, ref Param params)
             printf("%.*s", cast(int)data.length, data.ptr);
     }
 
+    if (OutBuffer* ob = params.makefileDeps)
+    {
+        foreach (i; 1 .. modules[0].aimports.dim)
+            semantic3OnDependencies(modules[0].aimports[i]);
+        Module.runDeferredSemantic3();
```

Comment:
Repeating my comment from #6961: You probably don't want to reuse -deps functionality as it is no longer useful for make. Analysing code via semantic3OnDependencies also reports local imports that are not used when compiling the code without the -deps option. It can also slow down compilation considerably. See discussion in #6748.

Reply:
I copied deps because it's what already does the job - otherwise I have no idea how to implement this.

Reply:
-deps does a lot more than just make file dependencies since #6748. What might work is to collect accessed files in `dmd.root.file.File.read`.